System/InputScheme.cpp

```diff
@@ -0,0 +1,360 @@
+#include "InputScheme.h"
+#include "allegro.h"
+
+namespace RTE {
+
+	const std::string InputScheme::c_ClassName = "InputScheme";
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void InputScheme::Clear() {
+		m_ActiveDevice = DEVICE_KEYB_ONLY;
+		m_SchemePreset = PRESET_NONE;
+		m_JoystickDeadzoneType = DeadZoneType::CIRCLE;
+		m_JoystickDeadzone = 0.01F;
+
+		for (int mapping = 0; mapping < INPUT_COUNT; ++mapping) {
+			m_InputMapping[mapping].Reset();
+		}
+
+		// Set up the default mouse button bindings - do these here because there are no config steps for them yet
```

Comment:
Don't know. I yeeted this without any noticeable ill effects but my guess is this was to make sure mouse works at whatever point when it couldn't be configured or something like that.
Works without it either way.

Reply:
Oh derp, I get it now. Yeah, I guess it's saying that we have no way to bind mouse button 1 and mouse button 2 in the options menu ingame (I assume we don't but I don't really remember and haven't checked).
Have you tested your later changes with mouse + keyboard and had it work fine?

Reply:
I deleted my settings file and let it rebuild on boot and everything was working fine because defaults are being assigned.
And I believe that since HardSphere made resetting the control schemes use default presets we can no longer get into a situation where there are no bindings and input doesn't work.

Reply:
Ahh, that makes sense. Cool.